scripts/screeps_rl_experiment_card.py

```
@@ -551,6 +585,176 @@ def validate_card(raw: Any) -> None:
+        except CardValidationError:
+            continue
+        if not isinstance(payload, dict) or payload.get("ok") is not True:
+            continue
+        run_id = accepted_dataset_run_id(payload)
+        if run_id is None:
+            continue
+        created_at = accepted_dataset_created_at(payload)
+        mtime = path.stat().st_mtime
+        candidates.append((created_at or "", mtime, run_id, path))
+    if not candidates:
+        raise CardValidationError(f"no accepted dataset gate with datasetRunId found under {gate_root}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)